app/routes/task_routes.py

```diff
@@ -1 +1,124 @@
+def create_new_task():
+    request_body = request.get_json()
+    if (
+        not request_body
+        or "title" not in request_body
+        or "description" not in request_body
+    ):
+        abort(make_response({"details": "Invalid data"}, 400))
+    new_task = Task.from_dict(request_body)
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {
+            "task":{
+            "id" : new_task.id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": False
+            }
+        }, 201
+
+@bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    sort_param = request.args.get("sort")
+
+    if sort_param == "desc":
+        query = db.select(Task).order_by(desc(Task.title))
+    else:
+        query = db.select(Task).order_by(asc(Task.title))
+    
+    tasks = db.session.scalars(query)
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+
+    return tasks_response
+
+@bp.get("/<id>")
+
+def get_one_task(id):
+    task = validate_model(Task, id)
+    
+    response = {        
+            "id" : task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+    if task.goal_id is not None:
+        response["goal_id"] = task.goal_id
+    return {"task": response}
+
+@bp.put("/<id>")
+def update_task(id):
+    task = validate_model(Task, id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    task.completed_at = request_body.get("is_complete",None)
+ 
+    db.session.commit()
+
+    return Response(status = 204, mimetype = "application/json")
+
+@bp.patch("/<id>/mark_complete")
+def mark_task_complete(id):
+    task = validate_model(Task, id)
+    if not task.completed_at:
+        task.completed_at = datetime.utcnow()
+
+        # Save the task updates in the database
+        # Uncomment the following line if you want to test the Slack API, 
+        # so that completed_at won't be stored and you can enter the if statement
+        db.session.commit()
+        
+        slack_api_key = os.environ.get('SLACK_BOT_API')
+        slack_channel = "test-slack-api"
+        slack_message = f"Someone just completed the task {task.title}"
+
+        if slack_api_key:
+            headers = {
+                "Authorization": f"Bearer {slack_api_key}"
+            }
+   
+            query_param = f"channel={slack_channel}&text={slack_message}"
+            
+            # Send a POST request to the Slack API
+            requests.post(f"https://slack.com/api/chat.postMessage?{query_param}", headers=headers)
+            
+    return "", 204
+
+@bp.patch("/<id>/mark_incomplete")
+def mark_task_incomplete(id):
+    task = validate_model(Task, id)
+    task.completed_at = None
+
+    db.session.commit()
+    return "", 204
```

Comment:
We don't have any tests that check that the response object returned from this route has the mimetype set as "application/json" so all of your unit tests pass. 

However, when you return `""`, the response object's mimetype is default set to "text/html". This means that all of your routes return JSON and this one returns HTML. 

We should prefer consistency so all of our routes are predictable. 

Instead of returning an empty string, we should create a Response object and explicitly set the "mimetype", like you do on line 124 for the delete route
```suggestion
    return Response(status = 204, mimetype = "application/json")
```